java/src/main/java/ai/rapids/cudf/ChunkedPack.java

```diff
@@ -88,7 +88,11 @@ public PackedColumnMetadata buildMetadata() {
 
   @Override
   public void close() {
-    chunkedPackDelete(nativePtr);
+    try {
+      chunkedPackDelete(nativePtr);
+    } finally {
+      nativePtr = 0;
```

Comment:
`finally` is always executed right? So what if `chunkedPackDelete` failed? Will `nativePtr` be orphaned?

Reply:
The problem is that this is a destructor throwing, and we don't know how far along in the destruction it progressed before stopping.  If we get an exception while trying to delete a native resource, we don't know whether it's safe to try to delete it again (and it's very likely to throw again if we try again).  That's why we null it out, since we explicitly do not want to try to use this native resource again, even for destruction.  The other wrapper classes do the same thing.

Reply:
But this `try` block will contain the exception within it. So if there was exception, should we rethrow it to let the application know that there was something wrong, instead of just keeping things silent like this?

Reply:
finally does not prevent the exception being thrown. It's important to keep the finally block free of exceptions which is the case here.

Reply:
I see. Thought that the `try{}` block always contains exceptions.

Reply:
> I see. Thought that the `try{}` block always contains exceptions.

only if it matches a `catch` block